src/main.ts

```diff
@@ -100,13 +103,33 @@ const executeDockerCommand = async (dockerCommand: string, continueOnError: bool
     var execOptions: any = {
         outStream: new NullOutstreamStringWritable({ decodeStrings: false }),
         listeners: {
-            stdout: (data: any) => console.log(data.toString()), //to log the script output while the script is running.
-            errline: (data: string) => {
+            stdout: async (data: any) => {
+                let scannedResult = {result: null};
+                if(!config.credScanEnable) console.log(data.toString());
+                else if(!process.env.CREDSCAN) console.log(data.toString());
+                else if(config.credScanEnable || process.env.CREDSCAN){
+                    await cs.credscan(data.toString(), scannedResult);
+                    if(scannedResult.result)
+                    console.log(scannedResult.result);
+                    else console.log(data.toString());
+                }
+                else console.log(data.toString());
+            }, //to log the script output while the script is running.
+            errline: async (data: string) => {
                 if (!shouldOutputErrorStream) {
                     errorStream += data + os.EOL;
                 }
                 else {
-                    console.log(data);
+                    let scannedResult = {result: null};
+                    if(!config.credScanEnable) console.log(data);

```

Comment:
This is common code, used above also, you can put it into one function.

Reply:
added a new function to main.ts for printing